website/docs/r/compute_instance_iam.html.markdown

```diff
@@ -0,0 +1,126 @@
+---
+layout: "google"
+page_title: "Google: google_compute_instance_iam"
+sidebar_current: "docs-google-compute-instance-iam"
+description: |-
+ Collection of resources to manage IAM policy for a GCE instance.
+---
+
+# IAM policy for GCE instance
+
+~> **Warning:** These resources are in beta, and should be used with the terraform-provider-google-beta provider.
+See [Provider Versions](https://terraform.io/docs/providers/google/provider_versions.html) for more details on beta resources.
+
+Three different resources help you manage your IAM policy for GCE instance. Each of these resources serves a different use case:
+
+* `google_compute_instance_iam_policy`: Authoritative. Sets the IAM policy for the instance and replaces any existing policy already attached.
+* `google_compute_instance_iam_binding`: Authoritative for a given role. Updates the IAM policy to grant a role to a list of members. Other roles within the IAM policy for the instance are preserved.
+* `google_compute_instance_iam_member`: Non-authoritative. Updates the IAM policy to grant a role to a new member. Other members for the role for the instance are preserved.
+
+~> **Note:** `google_compute_instance_iam_policy` **cannot** be used in conjunction with `google_compute_instance_iam_binding` and `google_compute_instance_iam_member` or they will fight over what your policy should be.
+
+~> **Note:** `google_compute_instance_iam_binding` resources **can be** used in conjunction with `google_compute_instance_iam_member` resources **only if** they do not grant privilege to the same role.
+
+## google\_compute\_instance\_iam\_policy
+
+```hcl
+data "google_iam_policy" "admin" {
+  binding {
+    role = "roles/editor"
+
+    members = [
+      "user:jane@example.com",
+    ]
+  }
+}
+
+resource "google_compute_instance_iam_policy" "instance" {
+	instance_name  = "your-instance-id"
+	policy_data = "${data.google_iam_policy.admin.policy_data}"
+}
+```
+
+## google\_compute\_instance\_iam\_binding
+
+```hcl
+resource "google_compute_instance_iam_binding" "instance" {
+  instance_name = "your-instance-id"
+  role       = "roles/compute.networkUser"
```

Comment:
Is `networkUser` a role that would make sense for an instance iam binding? Maybe change this to something that is specific to compute instance. Or generic owner/editor/viewer